test_runner/regress/test_tenants.py

```diff
@@ -201,3 +202,39 @@ def get_ps_metric_samples_for_tenant(tenant_id: TenantId) -> List[Sample]:
 
         post_detach_samples = set([x.name for x in get_ps_metric_samples_for_tenant(tenant)])
         assert post_detach_samples == set()
+
+
+def test_pageserver_with_empty_tenants(neon_simple_env: NeonEnv):
+    env = neon_simple_env
+    client = env.pageserver.http_client()
+
+    tenant_without_timelines_dir = env.initial_tenant
+    shutil.rmtree(Path(env.repo_dir) / "tenants" / str(tenant_without_timelines_dir) / "timelines")
+
+    tenant_with_empty_timelines_dir = client.tenant_create()
+    for timeline_dir_entry in Path.iterdir(
+        Path(env.repo_dir) / "tenants" / str(tenant_with_empty_timelines_dir) / "timelines"
+    ):
+        if timeline_dir_entry.is_dir():
+            shutil.rmtree(timeline_dir_entry)
+        else:
+            timeline_dir_entry.unlink()
+
```

Comment:
Why do we remove these directories manually?
If this is an expected state of pageserver, let's reach it using "legal" commands.

please, add a comment here as well

Reply:
Have added logging instead, so if something breaks, we could use logged ids to investigate.